src/main/scala/uk/gov/nationalarchives/consignmentexport/ChecksumValidator.scala

```diff
@@ -0,0 +1,24 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.util.UUID
+
+import cats.effect.IO
+import gov.loc.repository.bagit.domain.Bag
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import uk.gov.nationalarchives.consignmentexport.Validator.ValidatedFileMetadata
+
+import scala.jdk.CollectionConverters._
+
+class ChecksumValidator()(implicit val logger: SelfAwareStructuredLogger[IO]) {
+
+  def validateFileChecksums(bag: Bag, validatedFileMetadata: List[ValidatedFileMetadata]): IO[List[UUID]] = {
```

Comment:
Now that the method just does the comparison and doesn't throw an exception, a method name like `findMismatchedChecksums` might be clearer. `validate...` makes it sound like it's performing an action and doesn't return anything.

Reply:
renamed